WORLDSERVICE-543: Refactor CTAs (#12603)
bbc/simorgh: src/app/components/CallToActionLinkComponent/README.md

# Call to Action Link

## Description

A component following the `Compound Component` pattern to build Call to Action links.
The Compound Component pattern can be found inside the `index.tsx` entry point with the following Primitive Components:

> ### Components in this folder:
>
> - ButtonLikeWrapper
> - Text
> - Chevron

> ### Components outside this folder:
>

> > - Text

Reviewer: This is confusing - `Text` is in and outside this folder 😭 

It's perhaps best to just remove the "Components outside this folder" section.

Author: Ah yep

| htmlAttributes?    | string                         | Permits custom html attributes to be applied. E.g. `data-ignore-lite` for .lite                                                                                                                                                                                              |
| eventTrackingData? | shape({componentName: string}) | Tracking data needed for ati to track view and click of the Call to Action Link                                                                                                                                                                                              |
| download?          | boolean                        | Permits download                                                                                                                                                                                                                                                             |

> | fontVariant?       | string                         | Used to render the correct `font-family`, `font-style` and `font-weight`. Possible values are `sansRegular`, `sansRegularItalic`, `sansBold`, `sansBoldItalic`, `sansLight`, `serifRegular`, `serifMedium`, `serifMediumItalic`, `serifBold`, `serifLight`.                  |

Reviewer: Do we need to list all of the values here - could we link to where these are defined instead? (Somewhere within the themeprovider?)

> | size?              | string                         | The [BBC GEL size](https://www.bbc.co.uk/gel/features/typography#type-sizes). Possible values are `atlas`, `elephant`, `imperial`, `royal`, `foolscap`, `canon`, `trafalgar`, `paragon`, `doublePica`, `greatPrimer`, `bodyCopy`, `pica`, `longPrimer`, `brevier`, `minion`. |

Reviewer: As above, instead of listing them here, should we link to where they are defined as a type.

## Example ltr/rtl

> ### Call to Action Link with Flex Wrapper, Text and Chevron

Reviewer: Do we have a corresponding story for each of these?

Reviewer: Also:

```suggestion
### Call to Action Link with Button-like Wrapper, Text and Chevron
```

Author: I've reordered the README to match the stories. The component structure are the same. The stories have more props (e.g. alignWithMargin). But I haven't included these in the README to help keep it simple.

```javascript
<CallToActionLink url="https://www.bbc.com/ws/languages">
  <CallToActionLink.ButtonLikeWrapper>
    <CallToActionLink.Text>
      I am a link
      <CallToActionLink.Chevron />
    </CallToActionLink.Text>
  </CallToActionLink.ButtonLikeWrapper>
</CallToActionLink>
```

### Call to Action Link with Text and Chevron

> The Chevron can be inside or outside of the Text component.

Reviewer: If we're stating this, should we add an example of each?

  <CallToActionLink.Text>I am a link</CallToActionLink.Text>
  <CallToActionLink.Chevron />
</CallToActionLink>
```

### Call to Action Link with Text

```javascript
<CallToActionLink
  url="https://www.bbc.com/ws/languages"
  alignWithMargin
  size="brevier"
  fontVariant="sansRegular"
>
  <CallToActionLink.Text>I am a link</CallToActionLink.Text>
</CallToActionLink>
```

## Styles

The `alignWithMargin` prop applies `display: 'inline-block'` styles. Otherwise the link renders inline by default.

Interaction state (`hover`, `focus` and `visited`) styles are applied by default to ensure the component is accessible out of the box.

The css prop can be used to apply further styles, such as spacing and colours. These can be applied to interaction states. For example:

```javascript
<CallToActionLink
  url={link}
  className="focusIndicatorInvert"
  eventTrackingData={eventTrackingData}
  css={styles.callToActionLink}
>
  <CallToActionLink.ButtonLikeWrapper>
    <CallToActionLink.Text>
      {linkText}
      <CallToActionLink.Chevron />
    </CallToActionLink.Text>
  </CallToActionLink.ButtonLikeWrapper>
</CallToActionLink>
```

> ```javascript

Reviewer: Shall we mention that this is within an index.styles.ts file?

> calltoactionlink: ({palette}: Theme) =>

Reviewer: ```suggestion
callToActionLink: ({ palette }: Theme) =>
```

# Call to Action Link

## Description

A component following the `Compound Component` pattern to build Call to Action links.
The Compound Component pattern can be found inside the `index.jsx` entry point with the following Primitive Components:

> ### Components in this folder:
>
> - FlexWrapper
> - Text
> - Chevron

> > ### Components outside this folders:

Reviewer: ```suggestion
> ### Components outside this folder:
```
